src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -0,0 +1,277 @@
+    {
+        $this->block($message, 'OK', 'fg=white;bg=green', ' ', true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function error($message)
+    {
+        $this->block($message, 'ERROR', 'fg=white;bg=red', ' ', true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warning($message)
+    {
+        $this->block($message, 'WARNING', 'fg=white;bg=red', ' ', true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function note($message)
+    {
+        $this->block($message, 'NOTE', 'fg=yellow', ' ! ');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function caution($message)
+    {
+        $this->block($message, 'CAUTION', 'fg=white;bg=red', ' ! ', true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function table(array $headers, array $rows)
+    {
+        $table = new Table($this);
+        $table->setHeaders($headers);
+        $table->setRows($rows);
+        $table->setStyle('symfony-style-guide');
+
+        $table->render();
+        $this->newLine();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ask($question, $default = null, $validator = null)
+    {
+        $question = new Question($question, $default);
+        $question->setValidator($validator);
+
+        return $this->askQuestion($question, $validator);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function askHidden($question, $validator = null)
+    {
+        $question = new Question($question);
+        $question->setHidden(true);
+
+        return $this->askQuestion($question, $validator);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function confirm($question, $default = true)
+    {
+        return $this->askQuestion(new ConfirmationQuestion($question, $default));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function choice($question, array $choices, $default = null)
+    {
+        if (null !== $default) {
+            $values = array_flip($choices);
+            $default = $values[$default];
+        }
+
+        return $this->askQuestion(new ChoiceQuestion($question, $choices, $default));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function multipleChoice($question, array $choices, array $default = array())
+    {
+        $question = new ChoiceQuestion($question, $choices, $default);
+        $question->setMultiselect(true);
+
+        return $this->askQuestion($question);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function progress($max = 0)
+    {
+        $progress = parent::progress($max);
+        $progress->setEmptyBarCharacter('░'); // light shade character \u2591
+        $progress->setProgressCharacter('');
+        $progress->setBarCharacter('▓'); // dark shade character \u2593
+
+        return $progress;
+    }
+
+    /**
+     * @param Question $question
+     *
+     * @return string
+     */
+    public function askQuestion(Question $question)
```

Comment:
This method isn't in the interface, should I make it private or leave it public? My thinking for this being public was to allow the user to create a customized question (ie with autocomplete values).